### PR TITLE
Get rid of the using of the Function.prototype.bind method (fix #359)

### LINKED ISSUE #359: Javascript error on jsfiddle.net
See: https://testcafe-hhhm.devexpress.com/history?timestamp=1452867987933&siteUrl=https://jsfiddle.net/ \cc @miherlosev @LavrovArtem

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
/cc @VasilyStrelyaev please check test name
https://github.com/DevExpress/testcafe-hammerhead/pull/360/files#diff-daf554b967111c3bac68868c68540b3eR227

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/sandbox/code-instrumentation/getters-test.js [hunk 1]
```diff
@@ -222,3 +223,28 @@ test('get script body (T296958) (GH-183)', function () {
     if (typeof script.innerText === 'string')
         strictEqual(eval(processScript('script.innerText', true, false)).replace(/\s/g, ''), cleanedScriptCode.replace(/\s/g, ''));
 });
+
+test('function getAttributesProperty should correctly works if the Function.prototype.bind is removed (GH-359)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandbox/code-instrumentation/getters-test.js]:
```
test('the getAttributesProperty function should work correctly if Function.prototype.bind is removed (GH-359)', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

File: src/utils/fast-apply.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// OPTIMIZATION: http://jsperf.com/call-apply-optimization
```
Comment (inikulin) [inline src/utils/fast-apply.js]: It require client-server warning header

File: src/utils/create-func-wrapper.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default function createFuncWrapper (func, thisObj) {
```
Comment (inikulin) [inline src/utils/create-func-wrapper.js]: I'd prefer to use inline arrows instead of this

File: src/utils/fast-apply.js (hunk 2 above)
Comment (inikulin) [inline src/utils/fast-apply.js]: Nevermind. Seems like it's used only by client code. Let's move it to `src/client/utils/ then

**Comment by LavrovArtem** [comment]
FPR

File: src/client/utils/create-func-wrapper.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default (func, thisObj) => {
```
Comment (inikulin) [inline src/client/utils/create-func-wrapper.js]: This is not actually what I mean. But I was wrong. We don't need arrow here. And we should rename it to `fnBind` and we need a comment that clarifies why we need it.

File: src/client/utils/fast-apply.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// OPTIMIZATION: http://jsperf.com/call-apply-optimization
+export default (owner, methName, args) => {
```
Comment (inikulin) [inline src/client/utils/fast-apply.js]: No need for arrow here.

**Comment by inikulin** [comment]
\r-

**Comment by LavrovArtem** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by LavrovArtem** [comment]
ping @inikulin @churkin

**Comment by inikulin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by churkin** [comment]
lgtm

**Pull request merged**